### PR TITLE
Remove Data.setTokenExchangeUrl API from JavaScript and add the nullable check in Data.d.ts

### PR DESCRIPTION
**Pull request by YiMysty**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

A few sentences describing the overall goals of the pull request.

## Related PRs or issues

Change the null declaration to typescript style `?`

## Misc

Add what's missing, notes on what you tested, additional thoughts or questions.

### CODE REVIEW

File: appcenter-data/Data.d.ts [hunk 1]
```diff
@@ -23,13 +23,13 @@ export class WriteOptions {
 }
 
 export interface DocumentWrapper {
-    deserializedValue: object | null;
-    eTag: string | null;
+    deserializedValue?: object;
```
Comment (guperrot) [inline appcenter-data/Data.d.ts]: that's not the same: property being undefined or defined to be null.
did you check? I think we put null explicitly, please double check.

Comment (dhei) [reply appcenter-data/Data.d.ts]: Make sure you tested your assumption. 😄

Comment (YiMysty) [reply appcenter-data/Data.d.ts]: changed :)

File: appcenter-data/Data.d.ts [hunk 2]
```diff
@@ -48,17 +48,17 @@ export interface RemoteOperationListener {
...
 export interface RemoteOperationCompletedData {
-    id: string;
-    eTag: string;
-    partition: string;
+    id?: string;
+    eTag?: string;
+    partition?: string;
     operation: string;
-    error: Map<string, string> | null;
+    error?: Map<string, string>;
```
[the remark this answers is not in the document]
Comment (guperrot) [reply appcenter-data/Data.d.ts]: Well we only have 1 key for now which is message so technically it works as a map but this is too generic. The type is an Error that has a message.

Would be more like error?: DataError; where DataError has a single message property of type string.

[the remark this answers is not in the document]
Comment (YiMysty) [reply appcenter-data/Data.d.ts]: sure :D

[the remark this answers is not in the document]
Comment (dhei) [reply appcenter-data/Data.d.ts]: Discussed with @YiMysty, `DataSyncError` sounds more appropriate here. It's more generic and easier to follow.

File: appcenter-data/Data.d.ts [hunk 3]
```diff
@@ -47,18 +47,22 @@ export interface RemoteOperationListener {
...
-    error: Map<string, string> | null;
+    error?: DataSyncError;
 }
- 
+
 export function isEnabled(): Promise<boolean>;
 export function setEnabled(enabled: boolean): Promise<void>;
 export function setRemoteOperationListener(listener: RemoteOperationListener): Promise<void>;
-export function setTokenExchangeUrl(tokenExchangeUrl: string): void; 
+export function setTokenExchangeUrl(tokenExchangeUrl: string): void;
```
Comment (guperrot) [inline appcenter-data/Data.d.ts]: We don't expose that since it cannot work on Javascript (too late).
Needs to be called in native code.

Comment (guperrot) [reply appcenter-data/Data.d.ts]: We need to remove this from JS and bridge too, I think I told Jacky about it but this may have been lost.

Comment (dhei) [reply appcenter-data/Data.d.ts]: Remove this API from d.ts

Comment (YiMysty) [reply appcenter-data/Data.d.ts]: changed :)

**Pull request merged**